app/js/actions/jobs.js

```diff
@@ -1,13 +1,94 @@
-export const createJob = (job) => ({
-    type: 'CREATE_JOB',
+import actions from '../actions';
+import { makeB64Digest } from '../util/auth';
+
+// global variable to set and clear intervals for scanning for jobs.
+let jobInterval;
+
+export const newActiveJob = (workflow, job) => ({
+    type: 'NEW_ACTIVE_JOB',
+    workflow,
     job
 });
 
-export const startJob = (job) => ({
-    type: 'START_JOB',
+export const jobCompleted = (job) => ({
+    type: 'JOB_COMPLETED',
     job
 });
 
+export const pollJobStatus = () => {
+    return (dispatch, getState) => {
+        const { connection: { uri, availableApis } } = getState();
+        const url = `http://${uri.split('/')[0]}${availableApis[0]}jobs`;
+        fetch(url, {
+            method: 'GET'
+        })
+        .then((response) => {
+            if (!response.ok) {
+                throw Error(response.statusText);
+            }
+            return response.json();
+        })
+        .then((json) => {
+            if (json.completed.length !== 0) {
+                json.completed.map(job => {
+                    if (Object.keys(job).length !== 0) {
+                        dispatch(actions.jobCompleted(job));
+                        dispatch(actions.refreshArtifacts());
```

Comment:
This should be in the next promise, perhaps this one should return whether a reload is appropriate.

Also because there isn't any kind of "handshake" allowing the REST API to know that the UI finished consuming jobs we end up refreshing every artifact for every job in the list, current and past. We might need to refactor this into the reducers so that the interface is generally aware of what jobs it's seen before and what jobs it's looking for from the server.